java-checks/src/main/java/org/sonar/java/checks/MissingNewLineAtEndOfFileCheck.java

```diff
@@ -31,10 +31,16 @@
 
   @Override
   public void scanFile(JavaFileScannerContext context) {
-    String content = context.getFileContent();
-    char lastChar = content.charAt(content.length() - 1);
-    if( lastChar != '\n' && lastChar != '\r') {
+    if (isEmptyOrNotEndingWithNewLine(context.getFileContent())) {
```

Comment:
As discussed we don't want to break abstraction provided by Scanner. Therefore we read all file through their abstraction.